### PR TITLE
CI: Split coverage and leak build, fix leaks, remove aarch64 wheel build

### PR DESCRIPTION
**Pull request by fabratu**
This PR effectively fixes the broken coverage upload. Due to also testing for leaks and continuing on errors, the coverage for the C++ code way empty if a potential address leak is found. Both tests are now split into different runs. This should also speedup the coverage run due to having both leak checks and coverage at the same time.

In addition several potential leaks are fixed. One fix in `random.cc` from libstd is not fixed, since it is intentional. The fix instead is to remove the out-of-bound sanitizer checks for this file by adding a filter.

### CODE REVIEW

File: networkit/cpp/community/LouvainMapEquation.cpp [hunk 1]
```diff
@@ -34,6 +34,21 @@ LouvainMapEquation::LouvainMapEquation(const Graph &graph, bool hierarchical, co
     result = Partition(graph.upperNodeIdBound());
 }
 
+LouvainMapEquation::LouvainMapEquation(const Graph &graph, bool hierarchical, count maxIterations,
+                                       std::string_view parallelizationStrategy)
+    : CommunityDetectionAlgorithm(graph), hierarchical(hierarchical), maxIterations(maxIterations),
+      clusterCut(graph.upperNodeIdBound()), clusterVolume(graph.upperNodeIdBound()) {
+    parallelizationType = convertStringToParallelizationType(parallelizationStrategy);
```
Comment (bernlu) [inline networkit/cpp/community/LouvainMapEquation.cpp:41]: this line should be part of the initializer list and not in the function body

Comment (bernlu) [reply networkit/cpp/community/LouvainMapEquation.cpp:41]: The other variables in this function could be moved into the initializer list as well

File: networkit/cpp/generators/ChungLuGeneratorAlamEtAl.cpp [hunk 2]
```diff
@@ -121,10 +123,13 @@ Graph ChungLuGeneratorAlamEtAl::generateSequential() {
             const double p_j = static_cast<double>(groups[i].degrees)
                                * static_cast<double>(groups[j].degrees)
                                / static_cast<double>(sum_deg);
-            edgeSkipping(rng, addEdge, i, j, p_j, groups[i].vertexCount * groups[j].vertexCount);
+            if (p_j > 0.0)
```
Comment (bernlu) [inline networkit/cpp/generators/ChungLuGeneratorAlamEtAl.cpp:126]: maybe move this check into `edgeSkipping` instead

File: networkit/cpp/graph/TopologicalSort.cpp [hunk 3]
```diff
@@ -36,7 +34,7 @@ void TopologicalSort::checkDirected() {
 void TopologicalSort::checkNodeIdMap() {
     size_t numberOfNodes = G.numberOfNodes();
     std::vector<bool> checkTable(numberOfNodes);
-    for (auto &[origNode, mappedNode] : nodeIdMap) {
+    for (auto &[origNode, mappedNode] : *nodeIdMap) {
```
Comment (bernlu) [inline networkit/cpp/graph/TopologicalSort.cpp:37]: `nodeIdMap` could be a `nullptr` - check?

**Review by bernlu — CHANGES_REQUESTED** [review]

**Review by bernlu — DISMISSED** [review]

**Review by bernlu — DISMISSED** [review]

**Comment by fabratu** [comment]
Sry for the drive-by fix, but since the PR is finished and concerned with CI anyways, I added a missing package for the coverage build.

**Review by bernlu — APPROVED** [review]
